OWL2/Function.hs

```diff
@@ -65,31 +67,43 @@ instance Function PrefixMap where
 instance Function IRI where
   function a m qn = case m of
     StringMap pm -> case a of
-        Rename -> let pre = namePrefix qn in
-            qn { namePrefix = Map.findWithDefault pre pre pm}
-        Expand ->
-            let np = namePrefix qn
-                lp = localPart qn
-                iri = case iriType qn of
-                    Full -> let
-                      ex = np ++ ":" ++ lp
-                      res = qn {expandedIRI = ex}
-                      in if elem np ["http", "https"] then -- abbreviate
+     Rename -> let pre = prefixName qn in
+              qn { prefixName = Map.findWithDefault pre pre pm}
+     Expand ->
+      let np = prefixName qn
+          lp = show $ iriPath qn
+          iRi = if hasFullIRI qn then let
+                  ex = np ++ ":" ++ lp
+                  res = let x = expandCurie (Map.map mkIRI pm) qn in
+                         case x of
+                          Just y -> y
+                          Nothing -> error $ "could not expand:" ++ showIRI qn 
+                in if elem np ["http", "https"] then -- abbreviate
                         case Map.lookup "" pm of
                           Just ep | length ep > 5 -> case stripPrefix ep ex of
                             Just rl@(_ : _) -> res
-                              { namePrefix = ""
-                              , localPart = rl
-                              , iriType = Abbreviated }
+                              { prefixName = ""
+                              , iriPath = stringToId rl -- todo: maybe we should keep the Id structure of iriPath qn
```

Comment:
Maybe the "todo" needs to be addressed.

Reply:
@tillmo: I'm not sure about this one.

Reply:
This should be left for #1820. I have annotated the todo accordingly.